packages/@aws-cdk/aws-cognito/test/user-pool-client.test.ts

```diff
@@ -175,6 +175,30 @@ describe('User Pool Client', () => {
     });
   });
 
+  test('callbackUrls are not rendered if OAuth is disabled ', () => {
+    // GIVEN
+    const stack = new Stack();
+    const pool = new UserPool(stack, 'Pool');
+
+    // WHEN
+    new UserPoolClient(stack, 'PoolClient', {
+      userPool: pool,
+      disableOAuth: true,
+    });
+
+    // THEN
+    expect(stack).not.toHaveResourceLike('AWS::CognitoUserPoolClient', {
+      CallbackURLs: ['https://example.com'],
+    });
+
+    expect(stack).toHaveResource('AWS::Cognito::UserPoolClient', {
+      AllowedOAuthFlowsUserPoolClient: false,
+      SupportedIdentityProviders: [
+        'COGNITO',
+      ],
+    });
```

Comment:
This test can be made a bit more robust. As it is now, it will pass if the logic you're adding is reverted and the default is changed. Something like this instead would be more accurate - 

```suggestion
    // WHEN
    new UserPoolClient(stack, 'PoolClient', {
      userPool: pool,
      disableOAuth: true,
      callbackUrls: [ 'https://some-other-example.com' ],
    });

    // THEN
    expect(stack).toHaveResourceLike('AWS::CognitoUserPoolClient', {
      CallbackURLs: ABSENT,
    });
```

Reply:
neat, didn't know about `ABSENT`, will work it in. 

`callbackUrls` are a property contained in the `OAuthSettings` so the test wouldn't quite have that example. Adding the `oAuth` property would again run into validation because we don't allow `oAuth` to be specified if `disableOauth` is set to true.

nonetheless, updating the test!